### DISCUSSION TITLE
[question] how to watch for a CRD in any namespace?

### DISCUSSION THREAD

**Discussion by lknite** [kubernetes-client/csharp#1338] [Q&A] [answered]
I'm able to use WatchNamespacedAsync on a generic, to watch a specific namespace, with the understanding that WatchAsync is watching the default namespace, but what if I want to watch for a CRD which may appear in any namespace?

Current code watches one namespace:
```
await foreach (var (type, item) in generic.WatchNamespacedAsync<CrdMyCrd>("namespace_here"))
{
    Console.WriteLine("");
    Console.WriteLine("(event) [" + type + "] " + plural + "." + group + "/" + version + ": " + item.Metadata.Name);
```

Is this possible without creating multiple listens (one to watch each namespace)?

**Comment by tg123** [comment]
is your crd namespaced?
or you just want to watch in multiple ns?

**Comment by lknite** [comment, marked as answer]
There are instances of the crd in different namespaces, so may appear in multiple namespaces.

1. I'd like to just watch all of the crds wherever they are on the cluster with one watch command similar to how one of the examples lets you watch all pods in all namespaces.
2. It would also be nice if after I started the watch, if someone created a namespace & then created an instance of the crd, that that would aso result in an ADD event.  However, if I need to also watch for namespaces to be created & restart the CRD watch I could implement that... though it seems like "watch for a crd in any namespace" is something that might already exist.

**Reply by tg123** [reply]
set ns to `""`

**Reply by lknite** [reply]
I wondered about that ... testing...

**Reply by lknite** [reply]
easy enough, works great, thank you!
